electron/IPC/codeGenerator.ts

```diff
@@ -53,183 +62,200 @@ export default codeGenearator;
-      actionRs = await buildCode(rmprojFile, (event: IProgressEvent) => {
-        const { type, ...details } = event;
+        if (actionRs.isSuccess) {
+            // Then, build project + install dependencies
+            actionRs = await buildCode(rmprojFile, (event: IProgressEvent) => {
+                const { type, ...details } = event;
 
-        if (details.log) {
-          sb.appendLine(details.log);
+                if (details.log) {
+                    sb.appendLine(details.log);
+                }
+
+                browserWindow.webContents.send(type, details as IProgressDetail);
+            });
         }
 
-        browserWindow.webContents.send(type, details as IProgressDetail);
-      });
+        // Actions are completed, let's Save error (if any) to the .code-metadata
+        if (!actionRs.isSuccess) {
+            const metaFilePath = path.join(
+                rmprojFile.folderPath,
+                StandardFolder.OutputCode,
+                StandardOutputFile.MetaData
+            );
+            const metaData = JSON.parse(await fileSystem.readFile(metaFilePath)) as ITestSuitesMetadata;
+            metaData.error = { message: actionRs.errorMessage ?? "Unknown error" };
+            await fileSystem.writeFile(metaFilePath, JSON.stringify(metaData, null, 2));
+        }
+    } catch (err) {
+        actionRs = { isSuccess: false, errorMessage: String(err) };
     }
 
-    // Actions are completed, let's Save error (if any) to the .code-metadata
-    if (!actionRs.isSuccess) {
-      const metaFilePath = path.join(rmprojFile.folderPath, StandardFolder.OutputCode, StandardOutputFile.MetaData);
-      const metaData = JSON.parse(await fileSystem.readFile(metaFilePath)) as ITestSuitesMetadata;
-      metaData.error = { message: actionRs.errorMessage ?? "Unknown error" };
-      await fileSystem.writeFile(metaFilePath, JSON.stringify(metaData, null, 2));
+    const ipcRs: IIpcGenericResponse<{ logFile: string }> = {
+        isSuccess: actionRs.isSuccess,
+        errorMessage: actionRs.errorMessage,
+    };
+
+    // Print log file
+    try {
+        sb.appendLine(`*** Finish at ${moment().format("MMMM Do YYYY, h:mm:ss a")}`);
+        sb.appendLine(JSON.stringify(ipcRs, null, 4));
+
+        const logsFolder = path.join(rmprojFile.folderPath, StandardFolder.Logs);
+        await fileSystem.createFoler(logsFolder);
+
+        const logFileName = `gen-code.${moment(startTime).format("YYYYMMDD_HHmmss")}.log`;
+        const logFilePath = path.join(logsFolder, logFileName);
+        await fileSystem.writeFile(logFilePath, sb.toString());
+
+        ipcRs.data = { logFile: path.join(StandardFolder.Logs, logFileName) };
+    } finally {
+        browserWindow.webContents.send("finish", ipcRs);
+    }
+}
+
+async function generateProjectMetadata(
+    browserWindow: BrowserWindow,
+    event: Electron.IpcMainEvent,
+    projectFile: IRmProjFile
+): Promise<IIpcGenericResponse<ITestSuitesMetadata>> {
```

Comment:
Rename `ITestSuitesMetadata` (in common) to `IProjectMetadata`.  Because later, this interface does not only contain `suites` info, but also the other